src/opera/instance/base.py

```diff
@@ -57,9 +57,10 @@ def tosca_id(self):
     def state(self):
         return self.attributes["state"].data
 
-    def set_state(self, state):
+    def set_state(self, state, write=True):
         self.set_attribute("state", state)
-        self.write()
+        if write:
+            self.write()
```

Comment:
For what is this new `write` parameter needed? Just to write the initial state?

Reply:
It is needed if we do not want to save topology state on disk, just keep it in memory.